src/components/Swap/SwapActionButton/useOnSubmit.ts

```diff
@@ -13,14 +13,15 @@ export default function useOnSubmit() {
   const [, setInputAmount] = useSwapAmount(Field.INPUT)
 
   return useCallback(
-    async (submit: () => Promise<TransactionInfo | undefined>): Promise<boolean> => {
-      let info: TransactionInfo | undefined
+    async (submit?: () => Promise<TransactionInfo | void>): Promise<boolean> => {
+      let info: TransactionInfo | void
       try {
-        info = await submit()
+        info = await submit?.()
       } catch (e) {
-        console.error('Failed to submit', e)
+        console.error(e)
+        return false
       }
-      if (!info) return false
+      if (!info) return true
```

Comment:
this change seems significant, but it's hard to me to fully understand what's going on here. (Except that I know I'd do something like invert a bool for testing and maybe forget to switch it back.) Confirming this is intentional.

Reply:
This was a bug that is now fixed - this now returns false if the submission fails, instead of throwing.